src/hosted-buttons/utils.js

```diff
@@ -220,3 +225,86 @@ export const buildHostedButtonOnApprove = ({
     });
   };
 };
+
+export function getFlexDirection({
+  layout,
+}: GetFlexDirectionArgs): GetFlexDirection {
+  return { flexDirection: layout === "horizontal" ? "row" : "column" };
+}
+
+export function shouldRenderSDKButtons(
+  fundingSources: $ReadOnlyArray<FundingSources>
+): boolean {
+  return Boolean(fundingSources?.length);
+}
+
+export function getButtonColor(
+  color: Color,
+  fundingSource: FundingSources
+): Color {
+  const colorMap = {
+    gold: {
+      paypal: "gold",
+      venmo: "blue",
+      paylater: "gold",
+    },
+    blue: {
+      paypal: "blue",
+      venmo: "silver",
+      paylater: "blue",
+    },
+    black: {
+      paypal: "black",
+      venmo: "black",
+      paylater: "black",
+    },
+    white: {
+      paypal: "white",
+      venmo: "white",
+      paylater: "white",
+    },
+    silver: {
+      paypal: "silver",
+      venmo: "blue",
+      paylater: "silver",
+    },
+  };
+
+  return colorMap[color][fundingSource];
+}
+
+export function buildButtonContainer({
+  flexDirection,
+  selector,
+}: BuildButtonContainerArgs) {
+  if (typeof selector !== "string") {
+    throw new TypeError("Selector must be a string");
+  }
+
+  if (selector.charAt(0) !== "#") {
+    throw new TypeError("Selector must be referring to an id");
+  }
+
+  const buttonContainer = document.createElement("div");
+
+  buttonContainer.setAttribute(
+    "style",
+    `display: flex; flex-wrap: nowrap; gap: 16px; flex-direction: ${flexDirection}`
+  );
+
+  const primaryButton = document.createElement("div");
+  primaryButton.setAttribute("id", `ncp-primary-button`);
```

Comment:
are we using these ids anywhere else?

Reply:
No , just here. This will also get refactored since NCP team will:

1. provide us with an div#id in their html response to add the buttons to
2. they won't be providing a `fundingSoruces` array. This container building logic will be refactored once we get the updated response from them.